scanners/android/utility.go

```diff
@@ -69,122 +119,328 @@ The location of the file is: %s`, localPropertiesPth)
 	return ""
 }
 
-// EnsureGradlew - Retuns the gradle wrapper path, or error if not exists
-func EnsureGradlew(buildGradleFile string) (string, error) {
-	projectDir := filepath.Dir(buildGradleFile)
+func checkGradlew(projectDir string) error {
 	gradlewPth := filepath.Join(projectDir, "gradlew")
-	if exist, err := pathutil.IsPathExists(gradlewPth); err != nil {
-		return "", err
-	} else if !exist {
-		return "", errors.New(`<b>No Gradle Wrapper (gradlew) found.</b> 
+	exist, err := pathutil.IsPathExists(gradlewPth)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return errors.New(`<b>No Gradle Wrapper (gradlew) found.</b> 
 Using a Gradle Wrapper (gradlew) is required, as the wrapper is what makes sure
 that the right Gradle version is installed and used for the build. More info/guide: <a>https://docs.gradle.org/current/userguide/gradle_wrapper.html</a>`)
 	}
+	return nil
+}
 
-	return FixedGradlewPath(gradlewPth), nil
+func ensureAndroidLicences(androidHome string, isLegacySDK bool) error {
+	if !isLegacySDK {
+		licensesCmd := command.New(filepath.Join(androidHome, "tools/bin/sdkmanager"), "--licenses")
+		licensesCmd.SetStdin(bytes.NewReader([]byte(strings.Repeat("y\n", 1000))))
+		if err := licensesCmd.Run(); err == nil {
+			return nil
+		}
+	}
+
+	licenceMap := map[string]string{
+		"android-sdk-license":           "8933bad161af4178b1185d1a37fbf41ea5269c55\n\nd56f5187479451eabf01fb78af6dfcb131a6481e",
+		"android-googletv-license":      "\n601085b94cd77f0b54ff86406957099ebe79c4d6",
+		"android-sdk-preview-license":   "\n84831b9409646a918e30573bab4c9c91346d8abd",
+		"intel-android-extra-license":   "\nd975f751698a77b662f1254ddbeed3901e976f5a",
+		"google-gdk-license":            "\n33b6a2b64607f11b759f320ef9dff4ae5c47d97a",
+		"mips-android-sysimage-license": "\ne9acab5b5fbb560a72cfaecce8946896ff6aab9d",
+	}
+
+	licencesDir := filepath.Join(androidHome, "licenses")
+	if exist, err := pathutil.IsDirExists(licencesDir); err != nil {
+		return err
+	} else if !exist {
+		if err := os.MkdirAll(licencesDir, os.ModePerm); err != nil {
+			return err
+		}
+	}
+
+	for name, content := range licenceMap {
+		pth := filepath.Join(licencesDir, name)
+
+		if err := fileutil.WriteStringToFile(pth, content); err != nil {
+			return err
+		}
+	}
+
+	return nil
 }
 
-// GenerateOptions ...
-func GenerateOptions(searchDir string) (models.OptionModel, models.Warnings, error) {
-	warnings := models.Warnings{}
+func installMissingAndroidTools(srcRoot, androidHome string) error {
+	gradlewPath := filepath.Join(srcRoot, "gradlew")
```

Comment:
srcRoot can be empty? since instead of: `gradlew [command]` you would need to run: `./gradlew [command]`